flex_model/core/wrapper.py

```diff
@@ -31,24 +32,80 @@
 
 
 class FlexModel(nn.Module):
-    """Wraps a PyTorch model and applies forward and backward hooks.
-
-    Class which wraps a singular PyTorch model and installs/uninstalls
-    forward and backward hooks. The use provides a list of modules and
-    associated hook functions for each as input. To generate activations
-    as output, the user calls the forward function of the wrapped model
-    or optionally from the FlexModel. Contains no additional state
-    besides metadata and hook functions.
-
-    Note: See `torch.distributed.fsdp` for pytorch model wrapper example.
-
-    Args:
-        module (nn.Module):
-            PyTorch module to be wrapped.
-        output_ptr (Dict[str, Tensor]):
-            Object where retrieved activation are bound to during execution of
-            the hook function.
+    """Wraps a Pytorch :code:`nn.Module` to provide an interface for various
+    model-surgery techniques. Most importantly, allows registration of
+    user-instantiated :class:`HookFunction` classes which perform
+    model-surgery.
+
+    :note: Supported features include:
+
+        * Registry, enabling and disabling of :class:`HookFunction` instances.
+        * Exposing global states to all :class:`Hookfunction` runtimes.
+        * Distributed orchestration of 1-D to 3-D parallelisms.
+        * Providing convenience functions for various attributes.
+
+    :var nn.Module module: The wrapped Pytorch :code:`nn.Module` to hook into.
+    :var hook_functions: Collection of :class:`HookFunction` instances keyed by
+        the module name to hook into.
+    :type hook_functions: Dict[str, HookFunction]:
+    :var output_ptr: Pointer to output dictionary provided by the user.
+        Activations will be streamed here on the rank0 process only. The
+        returned tensors will all be on CPU.
+    :type output_ptr: Dict[str, Tensor]
```

Comment:
Don't think these types are rendering properly.

Reply:
Yeah seems like `:var:` and `:type:` don't play well together.